File: backend/app/schemas/infer.py
from pydantic import BaseModel, ConfigDict, Field


class InferRequest(BaseModel):
    disease: str = Field(..., pattern="^(flu|dengue)$", description="flu hoặc dengue")
    iso3: str = Field(..., min_length=3, max_length=3, description="ISO 3166-1 alpha-3")
    features: dict[str, float] = Field(
        ...,
        description="Feature values theo tên trong *_features.json. Thiếu key → dùng 0.0.",
    )


class InferResponse(BaseModel):
    model_config = ConfigDict(protected_namespaces=())

    disease: str
    iso3: str

    # Regression output
    predicted_log: float
    predicted_cases: float

    # Classification output
    risk_level: str
    p_low: float
    p_med: float
    p_high: float

    # Feature lists (để frontend biết cần truyền gì)
    regressor_features: list[str]
    classifier_features: list[str]

File: frontend/src/api/infer.ts
import api from "./axios";

export interface InferRequest {
  disease: "flu" | "dengue";
  iso3: string;
  features: Record<string, number>;
}

export interface InferResponse {
  disease: string;
  iso3: string;
  risk_level: "Low" | "Medium" | "High";
  p_low: number;
  p_med: number;
  p_high: number;
  predicted_cases: number;
  predicted_log: number;
}

export async function runInference(payload: InferRequest): Promise<InferResponse> {
  const { data } = await api.post<InferResponse>("/infer", payload);
  return data;
}
